setup.py

```diff
@@ -157,25 +158,29 @@ def run(self):
 
     print(f"-- {name} building version: {version}")
 
-    requirements = ['Pillow', 'psutil', 'future', 'packaging', 'wheel']
+    requirements = ["Pillow", "psutil", "future", "packaging", "wheel"]
 
     setup(
         name=name,
         version=version,
-        description=
-        'TorchServe is a tool for serving neural net models for inference',
-        author='PyTorch Serving team',
-        author_email='noreply@noreply.com',
+        description="TorchServe is a tool for serving neural net models for inference",
+        author="PyTorch Serving team",
+        author_email="noreply@noreply.com",
         long_description=pypi_description(),
-        url='https://github.com/pytorch/serve.git',
-        keywords='TorchServe PyTorch Serving Deep Learning Inference AI',
+        url="https://github.com/pytorch/serve.git",
+        keywords="TorchServe PyTorch Serving Deep Learning Inference AI",
         packages=pkgs,
         cmdclass={
-            'build_frontend': BuildFrontEnd,
-            'build_plugins': BuildPlugins,
-            'build_py': BuildPy,
+            "build_frontend": BuildFrontEnd,
+            "build_plugins": BuildPlugins,
+            "build_py": BuildPy,
         },
         install_requires=requirements,
-        entry_points={'console_scripts': ['torchserve=ts.model_server:start',]},
+        extras_require={
+            "onnx": ["numpy", "onnx","onnx-runtime"],
+            "ipex": ["intel_extension_for_pytorch"],
+        },
```

Comment:
Sure, this optional dependency pattern is one I've seen used by Ray for example the idea is you could run

```
pip install torchserve[onnx]
pip install ipex
```

To get the right dependencies

Now as far as putting things in `requirements-dev.txt` my take is everything that needs to be tested should be there. We can otherwise potentially explore `requirements-test.txt` where there should be no size limits